app/src/main/java/com/omen273/crossLingo/GameActivity.kt

```diff
@@ -468,14 +478,78 @@ class GameActivity : AppCompatActivity(), CrosswordView.OnLongPressListener,
 
     override fun onCrosswordUnsolved(view: CrosswordView) {}
 
-    override fun onSelectionChanged(view: CrosswordView, word: Crossword.Word?, position: Int) {
+    private var hits = 0
+    private var activateOnMoveCursorToSolvedCellsMode = false
+
+    @kotlin.ExperimentalStdlibApi
+    private fun detectMoveCursorToSolvedCellsMode(selection: CrosswordView.Selectable?,
+                                                  puzzleCells: Array<Array<CrosswordView.Cell?>>,
+                                                  ch: Char,
+                                                  position: Int)
+    {
+        val startRow = selection?.word?.startRow
+        val startColumn = selection?.word?.startColumn
+        val row = selection?.row
+        val column = selection?.column
+        var solved = 0
+        if(startRow != null && startColumn != null && row != null && column != null) {
+            when (selection.word.direction) {
+                Crossword.Word.DIR_ACROSS -> {
+                    for( i in column - 1 downTo startColumn)
+                        if ( puzzleCells[row][i]?.isFlagSet(CrosswordView.Cell.FLAG_SOLVED) == true)
+                            ++solved
+                }
+                Crossword.Word.DIR_DOWN -> {
+                    for( i in row - 1 downTo startRow)
+                        if ( puzzleCells[i][column]?.isFlagSet(CrosswordView.Cell.FLAG_SOLVED) == true)
+                            ++solved
+                }
+            }
+        }
+
+        if (selection != null) {
+            if(solved > 0 && ch.uppercaseChar() ==
+                selection.word.cells[position - solved].chars[0] &&
+                ch.uppercaseChar() != selection.word.cells[position].chars[0]
+            )
+                ++hits
+        }
+
+        val MAX_HITS_NUMBER = 3
+        if(hits >= MAX_HITS_NUMBER && hits < MAX_HITS_NUMBER + 1 &&
+            !activateOnMoveCursorToSolvedCellsMode)
+        {
+            val builder = AlertDialog.Builder(this).
+            setMessage(R.string.change_print_mode_to_move_to_solved)
+                .setPositiveButton(R.string.yes) { _, _ ->
+                    SettActivity.writePrintToFilledCellsToConfig(this, true)
+                    crosswordView.moveSelectionToSolvedSquares = true}
+                .setNegativeButton(R.string.no) { _, _ ->
+                }.create()
+            builder.setCancelable(false)
+            builder.show()
+            activateOnMoveCursorToSolvedCellsMode = true
```

Comment:
suggestion will show only once, may be disable listener after that to not waste resources?